### ISSUE TITLE
Use TileMapLayer for Godot exports as opposed to deprecated TileMapLayer

### ISSUE THREAD

**Issue by csueiras** [mapeditor/tiled#4106]
**Describe the bug**
Tiled exports to Godot using the `TileMap` node which has been deprecated (https://docs.godotengine.org/en/stable/tutorials/migrating/upgrading_to_godot_4.3.html#id4). 

I think the change on the plugin is actually quite small, it currently produces this:

```
[node name="01" type="Node2D"]

[node name="TileMap" type="TileMap" parent="."]
tile_set = ExtResource("TileSet_0")
format = 2
layer_0/name = "Ground"
layer_0/tile_data =  ...
```

We want:
```
[node name="01" type="Node2D"]

[node name="Ground" type="TileMapLayer" parent="."]
tile_set = ExtResource("TileSet_0")
use_parent_material = true
tile_map_data = PackedByteArray("
```

**To Reproduce**
1. Export a map to Godot
2. Open map in Godot
3. See deprecation warning for the use of a `TileMap` node

**Expected behavior**
The export should not produce a deprecation warning.

**Specifications:**
 - OS: macOS
 - Tiled Version: 1.11.0

**Comment by dogboydog** [comment]
Just to note, this would be a breaking change for anyone that is using Godot 4.0-4.2. Not sure if the plugin can read the project file and determine the project version or not.

**Comment by csueiras** [comment]
Yeah I did a good amount of digging into this and even started writing my own export plugin purely in javascript but decided to abandon and use YATI instead. Its easier to make sure things all work well if its an import flow in Godot as opposed to this export mechanism. 

Either way the current export plugin can generate the necessary things to then have Godot itself migrate it to a TileMapLayer so that perhaps is wnough of a workaround for the time beinf for most people.

**Comment by gab-gil** [comment]
Happened to bump into the same issue. TileMap are deprecated so it should at least offer 2 export options, `Godot 4 Scene files - TileMap` and `Godot 4 Scene files - TileMapLayer`. For new projects, building the whole tilemap system on something that's deprecated isn't a good option

**Comment by YusefZahran** [comment]
Hello. I've noticed this issue hasn't had any activity in a while and I assume since it's still open it hasn't been addressed yet, and a quick test on my machine shows the issue's result unchanged.

If it's alright @bjorn can you assign me this? It should be a relatively easy fix and I have experience working with Godot so I can test it. 
Thanks!

**Comment by bjorn** [comment]
@YusefZahran Sounds good! I'm not sure if at this point this even needs to be optional?

**Comment by YusefZahran** [comment]
If you're referring to the earlier discussion of having 2 export options depending on godot's engine version, we could parse the `godot.project` file (which is a plain text config file) that is in the godot project folder, and check for the version the current project is using. Tiled already makes sure it's exporting in a godot project directory so were' halfway there.

If the detected version is before 4.3 (the version `TileMap` was deperecated) it will continue with the current output. If not, we'll export with the newer `TileMapLayer` node type file.
